shared/src/commonMain/kotlin/org/hyperskill/app/step_quiz/presentation/StepQuizActionDispatcher.kt

```diff
@@ -204,36 +210,44 @@ internal class StepQuizActionDispatcher(
         ) {
-            val currentSubscription =
-                subscriptionsInteractor.getCurrentSubscription()
-                    .getOrThrow()
+            coroutineScope {
+                val currentSubscription =
+                    subscriptionsInteractor.getCurrentSubscription()
+                        .getOrThrow()
 
-            val currentProfile =
-                currentProfileStateRepository
-                    .getState()
-                    .getOrThrow()
+                val currentProfile =
+                    currentProfileStateRepository
+                        .getState()
+                        .getOrThrow()
 
-            val attempt =
-                stepQuizInteractor
-                    .getAttempt(action.step.id, currentProfile.id)
-                    .getOrThrow()
+                val attemptDeferred = async { stepQuizInteractor.getAttempt(action.step.id, currentProfile.id) }
+                val gptCodeWithErrorsDeferred = async { generateGptCodeWithErrors(action.step, currentProfile) }
 
-            val submissionState =
-                getSubmissionState(attempt.id, action.step.id, currentProfile.id)
-                    .getOrThrow()
+                val attempt = attemptDeferred.await().getOrThrow()
+                val submissionState =
+                    getSubmissionState(attempt.id, action.step.id, currentProfile.id)
+                        .getOrThrow()
 
-            val problemsLimitReachedModalData = getProblemsLimitReachedModalData(currentProfile, currentSubscription)
+                val problemsLimitReachedModalData =
+                    getProblemsLimitReachedModalData(currentProfile, currentSubscription)
 
-            Message.FetchAttemptSuccess(
-                step = action.step,
-                attempt = attempt,
-                submissionState = submissionState,
-                isProblemsLimitReached = currentSubscription.isProblemsLimitReached,
-                problemsLimitReachedModalData = problemsLimitReachedModalData,
-                problemsOnboardingFlags = onboardingInteractor.getProblemsOnboardingFlags()
-            )
+                val gptCodeWithErrors = gptCodeWithErrorsDeferred.await().getOrNull()
+
+                InternalMessage.FetchAttemptSuccess(
+                    step = action.step,
+                    attempt = attempt,
+                    submissionState = submissionState,
+                    isProblemsLimitReached = currentSubscription.isProblemsLimitReached,
+                    problemsLimitReachedModalData = problemsLimitReachedModalData,
+                    problemsOnboardingFlags = onboardingInteractor.getProblemsOnboardingFlags(),
+                    gptCodeGenerationWithErrorsData = StepQuizFeature.GptCodeGenerationWithErrorsData(
+                        isEnabled = currentProfile.features.isMobileGptCodeGenerationWithErrorsEnabled,
+                        code = gptCodeWithErrors
+                    )
+                )
```

Comment:
I suggest to return StepQuizFeature.GptCodeGenerationWithErrorsData from `generateGptCodeWithErrors` function